test/AbstractTransitionController.spec.ts

```diff
@@ -0,0 +1,209 @@
+    });
+  });
+
+  describe('startLoopingAnimation', () => {
+    it('should start the looping animation with exact parameters', () =>{
+      const reset = false;
+      const label = TransitionId.LOOP_1;
+
+      const spy = sinon.spy(component.transitionController, 'setupTimeline');
+      expect(component.transitionController.startLoopingAnimation(label, reset)).to.be.undefined;
+      expect(spy).to.be.calledWithExactly(TimelineType.LOOPING, reset, label);
+      spy.restore();
+    });
+  });
+
+  describe('setupTimeline', () => {
+    it('should setup looping timeline with label', () => {
+      expect(component.transitionController.setupTimeline(TimelineType.LOOPING, false, TransitionId.LOOP_1)).to.be.undefined;
+    });
+
+    it('should setup looping timeline', () => {
+      expect(component.transitionController.setupTimeline(TimelineType.LOOPING, false)).to.be.undefined;
+    });
+
+    it('should setup in timeline with label', () => {
+      expect(component.transitionController.setupTimeline(TimelineType.IN, false, TransitionId[TransitionDirection.IN].TRANSITION_ID_1)).to.be.undefined;
+    });
+
+    it('should setup in timeline', () => {
+      expect(component.transitionController.setupTimeline(TimelineType.IN, false)).to.be.undefined;
+    });
+
+    it('should setup out timeline with label', () => {
+      expect(component.transitionController.setupTimeline(TimelineType.OUT, false, TransitionId[TransitionDirection.OUT].TRANSITION_ID_1)).to.be.undefined;
+    });
+
+    it('should setup out timeline', () => {
+      expect(component.transitionController.setupTimeline(TimelineType.OUT, false)).to.be.undefined;
+    });
+  });
+
+
+  describe('stopLoopingAnimation', () => {
+    it('should stop looping animation', () => {
+      expect(component.transitionController.stopLoopingAnimation()).to.be.undefined;
+    });
+  });
+
+
+  describe('getComponent', () => {
+    it('should try to get a component by string', () => {
+      expect(app.transitionController.getComponent('[data-component="child-component-a"]')).to.be.a('object');
+    });
+
+    it('should try to get a  component by Component', () => {
+      expect(component.transitionController.getComponent(component)).to.be.a('object');
+    });
+
+    it('should try to get a  component by Element', () => {
+      expect(component.transitionController.getComponent(component.element)).to.be.a('object');
+    });
+
+    it('should trhow error for getComponent ', () => {
+      expect(component.transitionController.getComponent.bind(component.transitionController)).to.throw('The requested component [undefined]' +
+        ' does not exist');
+    });
+  });
+
+  describe('getTimelineDurationForComponent', () => {
+    it('should try to getTimelineDurationForComponent by string', () => {
+      expect(app.transitionController.getTimelineDurationForComponent('[data-component="child-component-a"]')).to.be.a('number');
```

Comment:
Would be better if we can set up a timeline where the exact duration would be known, and compare with that.